src/components/TodoList.tsx

```diff
@@ -0,0 +1,44 @@
+import { useContext, useMemo } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodoContext } from './TodoContext';
+import { Status } from '../types/Status';
+
+export const TodoList: React.FC = () => {
+  const {
+    todos, tempTodo, filter, todoIdsWithLoader,
+  } = useContext(TodoContext);
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case Status.Active:
+        return todos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return todos.filter(todo => todo.completed);
+      case Status.All:
+      default:
+        return todos;
+    }
+  }, [filter, todos]);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
```

Comment:
Use semantic tags for list `ul` `li`. Don't forget to remove default `li` dots and `ul` paddings